UndertaleModTool/MainWindow.xaml.cs

```diff
@@ -1972,91 +1972,120 @@ private void MainTree_PreviewMouseRightButtonDown(object sender, MouseButtonEven
+                            code.LocalsCount = 1;
+                            Data.CodeLocals.Add(locals);
+                        }
+                        (obj as UndertaleScript).Code = code;
+                    }
+                    if ((obj is UndertaleCode) && (Data.GeneralInfo.BytecodeVersion > 14))
+                    {
+                        UndertaleCodeLocals locals = new UndertaleCodeLocals();
+                        locals.Name = (obj as UndertaleCode).Name;
+                        UndertaleCodeLocals.LocalVar argsLocal = new UndertaleCodeLocals.LocalVar();
+                        argsLocal.Name = Data.Strings.MakeString("arguments");
+                        argsLocal.Index = 0;
+                        locals.Locals.Add(argsLocal);
+                        (obj as UndertaleCode).LocalsCount = 1;
+                        Data.CodeLocals.Add(locals);
+                    }
+                }
+                else
+                {
+                    (obj as UndertaleNamedResource).Name = new UndertaleString(notDataNewName); // not Data.MakeString!
                 }
             }
+            else if (obj is UndertaleString str)
+                str.Content = "string" + list.Count;
+            list.Add(obj);
+            UpdateTree();
+            HighlightObject(obj);
+            OpenInTab(obj, true);
         }
-        private void MenuItem_OpenInNewTab_Click(object sender, RoutedEventArgs e)
-        {
-            OpenInTab(Highlighted, true);
-        }
-        private void MenuItem_FindAllReferences_Click(object sender, RoutedEventArgs e)
-        {
-            var obj = (sender as FrameworkElement)?.DataContext as UndertaleResource;
-            if (obj is null)
-            {
-                this.ShowError("The selected object is not an \"UndertaleResource\".");
-                return;
-            }
 
-            FindReferencesTypesDialog dialog = null;
-            try
-            {
-                dialog = new(obj, Data);
-                dialog.ShowDialog();
-            }
-            catch (Exception ex)
-            {
-                this.ShowError("An error occured in the object references related window.\n" +
-                               $"Please report this on GitHub.\n\n{ex}");
-            }
-            finally
-            {
-                dialog?.Close();
-            }
-        }
-        private void MenuItem_CopyName_Click(object sender, RoutedEventArgs e)
-        {
-            CopyItemName(Highlighted);
-        }
-        private void MenuItem_Delete_Click(object sender, RoutedEventArgs e)
+        private void MenuItem_AddType_Click(object sender, RoutedEventArgs e)
         {
-            if (Highlighted is UndertaleObject obj)
-                DeleteItem(obj);
-        }
+            //this 99% copied from the the Add_Click, so could probably have the fuctionality of it (and above) combined into one method that then gets called from both events)
```

Comment:
please don't copy paste for 90%

Also because I cant find it right now, whats the difference between Add and AddType ?